blog/_posts/2018-02-07-collections-performance.md

```diff
@@ -0,0 +1,136 @@
+  with IndexedSeqOps[A, Vector, Vector[A]]
+  with StrictOptimizedSeqOps[A, Vector, Vector[A]]
+~~~
+
+Here we use `StrictOptimizedSeqOps`, which is a specialization of `StrictOptimizedIterableOps`
+for `Seq` collections.
+
+## Is The View Based Design Worth It?
+
+In my previous article I explained that a drawback of the old builder based design was that,
+on non strict collections (e.g. `Stream` or `View`), we had to carefully override all the
+default implementations of transformation operations to make them non strict.
+
+Now it seems that the situation is just reversed: the default implementations work well
+with non strict collections, but we have to override them in strict collections.
+
+So, is the new design worth it? To answer this question I will quote a comment posted
+by Stefan Zeiger [here](https://www.reddit.com/r/scala/comments/7g52cy/let_them_be_lazy/dqixt8d/):
+
+> The lazy-by-default approach is mostly beneficial when you're implementing lazy
+> collections because you don't have to override pretty much everything or get
+> incorrect semantics. The reverse risk is smaller: If you don't override a lazy
+> implementation for a strict collection type you only suffer a small performance
+> impact but it's still correct.
+
+In short: implementations are **correct first** in the new design but you might want to
+override them for performance reasons on strict collections.
+
+## Performance Comparison With 2.12’s Collections
+
+Talking about performance, how performant are the new collections compared to the old ones?
+
+Again, the answer depends on the type of collection, the operations and the number of elements.
+My `Vector` benchmarks show a 20% speedup on average:
+
+![](/resources/img/new-collections-performance-filter.png)
+
+![](/resources/img/new-collections-performance-map.png)
+
+![](/resources/img/new-collections-performance-flatMap.png)
+
+These charts show the execution time (vertically) of the `filter`, `map` and `flatMap`
+operations, according to the number of elements (horizontally). Note that scales are
+logarithmic in both axis. The blue line shows the performance of the old `Vector`,
+the green line shows the performance of the new `Vector` if it used only view based
+implementations, and the red line shows the actual performance of the new `Vector`
+(with strict optimized implementations). Benchmark source code and numbers can be found
+[here](https://gist.github.com/julienrf/f1cb2b062cd9783a35e2f35778959c76).
+
+Since operation implementations end up being the same, why do we get better performance
+at all? Well, these numbers are specific to `Vector`, and are due to the fact that
+we more agressively inlined a few critical methods. I don’t expect the new collections
+to be *always* 20% faster than the old collections. However, there is no reason for
+them to be slower since the execution path, when calling an operation, can be made
+exactly the same as in the old collections.
+
+## Conclusion
```

Comment:
I would include "correct first" somewhere in the conclusion, it's an interesting point worth repeating.